### PR TITLE
Add diskImageURL to use the cache version of disk

### PR DESCRIPTION
**Pull request by praveenkumar**
Add diskImageURL to use the cache version of disk
    
     - Remove crc repo dependency to make it work with go module
     - Add sshkeypath to driver struct.
     - Add go module

### CODE REVIEW

File: libvirt.go [hunk 1]
```diff
@@ -6,40 +6,40 @@ import (
 	"encoding/xml"
 	"errors"
 	"fmt"
+	"github.com/code-ready/machine/libmachine/mcnutils"
```
Comment (gbraad) [inline libvirt.go]: wrong location... should be near line 19

**Review by gbraad — CHANGES_REQUESTED** [review]

File: libvirt.go [hunk 2]
```diff
@@ -159,82 +169,46 @@ func (d *Driver) validateNetwork() error {
...
-			log.Debugf("Reactivating network: %s", err)
-			err = network.Create()
-			if err != nil {
-				log.Warnf("Failed to Start network: %s", err)
-				return err
-			}
-		}
-		return nil
+	if err != nil {
+		return fmt.Errorf("Use 'crc setup' for define the network, %+v", err)
```
Comment (gbraad) [inline libvirt.go:173]: :+1:

File: libvirt.go [hunk 3]
```diff
@@ -159,82 +169,46 @@ func (d *Driver) validateNetwork() error {
...
-	network, err = conn.NetworkDefineXML(xml.String())
-	if err != nil {
-		log.Errorf("Failed to create network: %s", err)
-		return nil
-	}
-	err = network.SetAutostart(true)
-	if err != nil {
-		log.Warnf("Failed to set network to autostart: %s", err)
+	if nw.Ip.Address == "" {
+		return fmt.Errorf("%s network doesn't have DHCP configured properly", d.Network)
```
Comment (gbraad) [inline libvirt.go]: drop `properly` or 'properly configured'. however, the sentence is still kinda 'faux-detailed'.

File: libvirt.go [hunk 4]
```diff
@@ -264,6 +238,10 @@ func (d *Driver) PreCreateCheck() error {
 }
 
 func (d *Driver) Create() error {
+	b2dutils := mcnutils.NewB2dUtils(d.StorePath)
```
Comment (gbraad) [inline libvirt.go:241]: ugh... :-s

Comment (praveenkumar) [reply libvirt.go:241]: Yes this is the way I put the disk to respective directory, we can refactor that bit later.

**Review by gbraad — APPROVED** [review]

**Pull request merged**